org.eclipse.scanning.test/src/org/eclipse/scanning/test/event/DeviceRequestTest.java

```diff
@@ -219,19 +237,42 @@ public void testInvalidName() throws Exception {
 	}
 
 	@Test
-	public void testMandelModelConfigure() throws Exception {
-		
+	public void testMandelbrotDeviceInfo() throws Exception {
 		DeviceRequest req = new DeviceRequest();
 		req.setDeviceName("mandelbrot");
 		DeviceRequest res = requester.post(req);
-		
+
+		@SuppressWarnings("unchecked")
 		DeviceInformation<MandelbrotModel> info = (DeviceInformation<MandelbrotModel>)res.getDeviceInformation();
-		if (info==null) throw new Exception("There were no devices found and at least the mandelbrot example should have been!");
-		
+		assertNotNull("There were no devices found and at least the mandelbrot example should have been!", info);
+		assertEquals("Example mandelbrot device", info.getDescription());
+		assertEquals(DeviceRole.HARDWARE, info.getDeviceRole());
+		assertNull(info.getHealth()); // TODO what does this attribute mean?
+		assertEquals("Example Mandelbrot", info.getLabel());
+		assertEquals(1, info.getLevel());
+		assertEquals("mandelbrot", info.getName());
+		assertEquals(DeviceState.READY, info.getState());
+		assertEquals(new HashSet<>(Arrays.asList(ScanMode.SOFTWARE)), info.getSupportedScanModes());
+
+		IRunnableDevice<MandelbrotModel> mandy = dservice.getRunnableDevice("mandelbrot");
+		assertEquals(mandy.getModel(), info.getModel());
+	}
+
+	@Test
+	public void testMandelbrotConfigure() throws Exception {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)